server/controllers/posts.js

```diff
@@ -37,13 +38,15 @@ const savePost = asyncMiddleware(async (req, res) => {
 
 const getPostsByType = asyncMiddleware(async (req, res, next) => {
   const typesList = ['search', 'preferences', 'followings', 'popular']
-  const { type } = req.query
+  const {
+    type
+  } = req.query
```

Comment:
Besides "type" variable also de-structure "index, tags, emoji" which are used down the lines